include/swift/AST/Types.h

```diff
@@ -1397,26 +1399,22 @@ class BuiltinIntegerWidth {
   bool isArbitraryWidth() const { return RawValue == ArbitraryWidth; }
   
   /// Get the least supported value for the width.
-  ///
-  /// FIXME: This should be build-configuration-dependent.
   unsigned getLeastWidth() const {
     if (isFixedWidth())
       return getFixedWidth();
     if (isPointerWidth())
-      return 32;
+      return RawPointerSize;
```

Comment:
I don't think this is reasonable.  If we're going to rely on knowing the concrete width of a pointer, we should use that to eliminate the pointer-sized option from `BuiltinIntegerWidth` completely.  Otherwise we should keep it abstract and just change `getLeastWidth` to return 16 for it.

Reply:
I understand. I'll adjust the PR to get rid of this change and put pointers back to the abstract pointer type. The issue I was trying to fix was in `getCastOperation` in `Builtins.cpp`, it rules out cast operations that should in fact be valid (e.g. `Builtin.zextOrBitCast_Word_Int32(word)`, this should in fact lower correctly on 32 bit and 16 bit archs but reports as an error). But my fix was overkill and unreasonable. I'll modify just `getCastOperation` to allow it to understand target pointer size correctly and re-submit a cleaned up PR with this and the rest of the feedback. I'm happy if you guys don't even want this too. Will totally understand!